pkg/branchcuts/bumper/bumper.go

```diff
@@ -0,0 +1,106 @@
+package bumper
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/sirupsen/logrus"
+
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+)
+
+type Bumper[T any] interface {
```

Comment:
I don't think that you need to specify the type here, and I don't believe that you need the whole interface. Each of the tools uses those methods in a different way.  You could have the `Bump` as a function that gets a list of objects and destinations to bump.